pkg/credentials/jwt_source_provider.go

```diff
@@ -0,0 +1,17 @@
+package credentials
+
+import (
+	"github.com/spiffe/go-spiffe/v2/spiffeid"
+)
+
+type JWTSourceProvider func(subject spiffeid.ID, workloadSocket string, audience string) JWTSource
+
+func StandardJWTSourceProvider(subject spiffeid.ID, workloadSocket string, audience string) JWTSource {
+	return NewJWTSVIDSource(subject, workloadSocket, audience)
+}
+
+func StaticJWTSourceProvider(source JWTSource) func(spiffeid.ID, string, string) JWTSource {
+	return func(spiffeid.ID, string, string) JWTSource {
+		return source
+	}
+}
```

Comment:
JWTSourceProvider is a factory interface which allows us to acquire an instance of JWTSource. The other two functions are for convenience since the inline function definitions can get a bit crufty: the first is the "standard" logic (i.e. create the actual provider) and the second just returns the instance that's handed to it, primarily for testing.